### PR TITLE
exception-handling

### PR DESCRIPTION
**Pull request by harshit050499**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -10,7 +10,7 @@
 	</parent>
 	<groupId>com.mykaarma</groupId>
 	<artifactId>urlshortener</artifactId>
-	<version>0.0.1</version>
+	<version>0.0.2-RC1</version>
```
Comment (prakash-42) [inline pom.xml]: use snapshot, this is a JAR

File: src/main/java/com/mykaarma/urlshortener/service/UrlService.java [hunk 2]
```diff
@@ -142,14 +144,75 @@ public UrlDetails shortenUrl(String longUrl, String shortUrlDomain, long expiryD
...
 		}
 		
 		return shortUrlDetails;
 		
 	}
-	
+
+	private UrlDetails shortUrlRetry(String shortUrlDomain, String longUrl, Date expiryDate, String businessUUID, Map<String, String> additionalParams,String urlPrefix,int retryCount) {
+
+		log.info(String.format("Creating a new shortUrl using retry method  for longUrl=%s and businessUUID=%s", longUrl, businessUUID));
```
Comment (prakash-42) [inline src/main/java/com/mykaarma/urlshortener/service/UrlService.java:125]: @harshit050499 I see repeated code here, please make sure no code is repeated.

**Review by prakash-42 — DISMISSED** [review]

**Review by vishal2295 — DISMISSED** [review]

File: src/main/java/com/mykaarma/urlshortener/service/UrlService.java [hunk 3]
```diff
@@ -136,20 +146,34 @@ public UrlDetails shortenUrl(String longUrl, String shortUrlDomain, long expiryD
...
+
 		shortUrlDetails.setShortUrl(shortUrl);
-		
+
 		try {
 			urlRepository.saveUrl(shortUrlDetails);
+		}
+		catch (ShortUrlDuplicateException e)
+		{
+			log.info(String.format("Mongo Exception Occurred, retrying creating short url for longUrl=%s businessUUID=%s domainPurpose=%s",longUrl, businessUUID, shortUrlDomain));
```
Comment (prakash-42) [inline src/main/java/com/mykaarma/urlshortener/service/UrlService.java]: Don't use word `mongo` in logs, use something like DuplicateHashKeyException occurred, retrying ....

**Review by prakash-42 — DISMISSED** [review]

**Review by prakash-42 — DISMISSED** [review]

File: src/main/java/com/mykaarma/urlshortener/service/UrlService.java [hunk 4]
```diff
@@ -136,20 +146,34 @@ public UrlDetails shortenUrl(String longUrl, String shortUrlDomain, long expiryD
...
+
 		try {
 			urlRepository.saveUrl(shortUrlDetails);
+		}
+		catch (ShortUrlDuplicateException e)
+		{
+			log.info(String.format("DuplicateHashKeyException Exception Occurred, retrying creating short url for longUrl=%s businessUUID=%s domainPurpose=%s",longUrl, businessUUID, shortUrlDomain));
+			if(retryCount ==6)
+			{
+				log.error("Failed to create shortUrl even after retries for long url=%s",longUrl);
```
Comment (prakash-42) [inline src/main/java/com/mykaarma/urlshortener/service/UrlService.java]: @harshit050499 also log the exception in here. Also, please test this log, I don't think log.info will recognize `%s`

**Review by prakash-42 — DISMISSED** [review]

**Review by vishal2295 — APPROVED** [review]

**Pull request merged**